fiftyone/plugins/secrets.py

```diff
@@ -101,20 +104,35 @@ def _get_secrets_client():
     return client()
 
 
-class SecretsDictionary:
+class SecretsDictionary(Mapping[str, str]):
     """
     A more secure dictionary for accessing plugin secrets in
     operators that will attempt to resolve missing plugin secrets upon access.
     """
 
-    def __init__(self, secrets_dict, operator_uri=None, resolver_fn=None):
-        self.__secrets = secrets_dict  # types.MappingProxyType(secrets_dict)
+    def __init__(
+        self,
+        secrets_dict,
+        operator_uri=None,
+        resolver_fn=None,
+        required_keys=None,
+    ):
+        self.__secrets = secrets_dict
+        self.__required_keys = required_keys
         self._operator_uri = operator_uri
         if resolver_fn:
             self._resolver = resolver_fn
         else:
             self._resolver = None
 
+    def __len__(self):
+        """Returns the number of secrets defined in the plugin definition"""
+        return len(self.__required_keys)
+
+    def __iter__(self):
```

Comment:
For dicts, `__iter__` should only return keys, not values

Reply:
Fixing this is important because users often iterate over dicts like this:

```py
for key in ctx.secrets:
    value = ctx.secrets[key]
```

Reply:
fixed